pandas/_libs/ops.pyx

```diff
@@ -308,3 +309,20 @@ def maybe_convert_bool(ndarray[object] arr,
             return (arr, None)
     else:
         return (result.view(np.bool_), None)
+
+
+@cython.overflowcheck(True)
+cpdef int64_t calc_int_int(object op, int64_t a, int64_t b) except? -1:
+    """
+    Calculate op(a, b) and return the result. Raises OverflowError if converting either
+    operand or the result to an int64_t would overflow.
+    """
+    return op(a, b)
+
+@cython.overflowcheck(True)
+cpdef int64_t calc_int_float(object op, int64_t a, double b) except? -1:
```

Comment:
looks like this isn't currently needed outside of tests?  if so, let's avoid implementing it until it is needed

Reply:
Heh, jinx, I just deleted it (unpushed commit).